x/mongo/driver/operation.go

```diff
@@ -890,7 +922,7 @@ func (op Operation) addBatchArray(dst []byte) []byte {
 	return dst
 }
 
-func (op Operation) createQueryWireMessage(dst []byte, desc description.SelectedServer) ([]byte, startedInformation, error) {
+func (op Operation) createQueryWireMessage(maxTimeMS *int64, dst []byte, desc description.SelectedServer) ([]byte, startedInformation, error) {
```

Comment:
Negative `maxTimeMS` values are not valid and a 0 value means no server-side timeout (see docs [here](https://www.mongodb.com/docs/manual/reference/command/aggregate/)). Since there's no valid use case for a non-positive `maxTimeMS` value, we should change the `maxTimeMS` param type to a `uint64` and only include it in the request if it's positive.

Reply:
Great point; thanks for linking to the docs. Switched to `uint64` (really `*uint64`).